spark/src/main/scala/org/apache/spark/sql/delta/commands/merge/MergeOutputGeneration.scala

```diff
@@ -164,8 +165,8 @@ trait MergeOutputGeneration { self: MergeIntoCommandBase =>
     // That is, conditionally invokes them based on whether there was a match in the outer join.
 
     // Predicates to check whether there was a match in the full outer join.
-    val ifSourceRowNull = col(SOURCE_ROW_PRESENT_COL).isNull.expr
-    val ifTargetRowNull = col(TARGET_ROW_PRESENT_COL).isNull.expr
+    val ifSourceRowNull = expression(col(SOURCE_ROW_PRESENT_COL).isNull)
+    val ifTargetRowNull = expression(col(TARGET_ROW_PRESENT_COL).isNull)
```

Comment:
Why do we make this change sometimes? (using `expression(..)`)

Other times it seems we use the implicit conversion to `RichColumn` in `ClassicColumnConversions`?

Reply:
Is there a reason we need `expression(..)` at all? And cannot just leave all these instances

Reply:
`expression(...)` is much cheaper, and does not rely on the `SparkSession.active` thread local. The downside is that it can only be used in scenarios where you know the Expression is used to build a Column.

Reply:
Gotcha. Could you maybe add this in docs to `ColumnConversionShims`? It would be good to have this info somewhere since I don't think it'll be obvious to most developers